main/src/com/google/refine/operations/row/RowDuplicatesRemovalOperation.java

```diff
@@ -0,0 +1,130 @@
+/*
+
+Copyright 2024 OpenRefine
+All rights reserved.
+*/
+
+package com.google.refine.operations.row;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.google.refine.browsing.Engine;
+import com.google.refine.browsing.EngineConfig;
+import com.google.refine.browsing.FilteredRows;
+import com.google.refine.browsing.RowVisitor;
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.Column;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.model.changes.RowRemovalChange;
+import com.google.refine.operations.EngineDependentOperation;
+import com.google.refine.operations.OperationDescription;
+
+public class RowDuplicatesRemovalOperation extends EngineDependentOperation {
+
+    final protected List<String> _criteria;
+    final List<Column> criteriaColumns = new ArrayList<Column>();
+    HashSet<Object> rowUniqueKeys = new HashSet<>();
```

Comment:
The fields in an `Operation` class should only contain the operation metadata, not the data obtained from its execution. I would move this field into the `RowVisitor` since it's only needed there.

Reply:
Moved rowUniqueKeys to RowVisitor